extensions.html

```diff
@@ -334,6 +334,117 @@ <h2><dfn>GamepadPose</dfn> Interface</h2>
+          "rgb"
+        };
+      </pre>
+      
+      <dl data-dfn-for="GamepadLightIndicatorType">
+        <dt>"<dfn>on-off</dfn>"</dt>
+        <dd>
+          Light indicator that supports a single color.
+        </dd>
+        
+        <dt>"<dfn>rgb</dfn>"</dt>
+        <dd>
+          Light indicator that supports multiple colors.
+        </dd>
+      </dl>
+    </section>
+    
+    <section>
+      <h2><dfn>GamepadLightColor</dfn></h2>
+      <p>
+        This represents the color of a light indicator. The default value should
+        be set to 0 for red, green, and blue.
+      </p>
+    
+      <pre class="idl">
+        dictionary GamepadLightColor {
+          required octet red;
+          required octet green;
+          required octet blue;
+       };
+      </pre>
+      
+      <dl data-dfn-for="GamepadLightColor">
+        <dt><dfn>red</dt></dfn>
+        <dd>
+          Red component of the light color, or non-zero value for an
+          <code>on-off</code> light indicator that indicates ON.
+        </dd>
+        
+        <dt><dfn>green</dt></dfn>
+        <dd>
+          Blue component of the light color, or non-zero value for an
+          <code>on-off</code> light indicator that indicates ON.
+        </dd>
+        
+        <dt><dfn>blue</dt></dfn>
+        <dd>
+          Green component of the light color, or non-zero value for an
+          <code>on-off</code> light indicator that indicates ON.
+        </dd>
+      </dl>
+    </section>
+    
+    <section>
+      <h2><dfn>GamepadLightIndicator</dfn></h2>
+      <p>
+        This interface defines a light indicator.
+      </p>
+      
+      <pre class="idl">
+        [Exposed=Window, SecureContext]
+        interface GamepadLightIndicator {
+          readonly attribute GamepadLightIndicatorType type;
+        
+          Promise&lt;long&gt; setColor(GamepadLightColor color);
+      };
+      </pre>
+      
+      <dl data-dfn-for="GamepadLightIndicator">
+        <dt><dfn>type</dt></dfn>
+        <dd>
+          Type of light indicator supported by the gamepad.
+        </dd>
+        
+        <dt><dfn>setColor</dt></dfn>
+        <dd>
+          <p>
+          Sets the color of a light indicator, or sets the light indicator to ON
+          or OFF.
+          </p>
+          
+          <p>
+          For <code>on-off</code> light indicators, one or more non zero values
+          denotes the light indicator is ON, whereas a zero value for all
+          components indicates the light indicator is OFF.
+          </p>
+          
+          <p>
+          For RGB light indicators, one or more non zero values specify the
+          color of the light indicator, whereas all zero values indicate the
+          light indicator is OFF.
+          </p>
+          
+          <p>
+          The returned Promise will resolve <code>true</code> when setting the
+          color has succeeded, and will reject the device API error code on
+          failure.
```

Comment:
This needs to reject with a an actual DOM Exception of some kind. I think this is overly broad/underspecified tho...